zipline/algorithm.py

```diff
@@ -409,6 +410,12 @@ def noop(*args, **kwargs):
         if 'data_frequency' in kwargs:
             self.data_frequency = kwargs.pop('data_frequency')
 
+        capital_base = self.sim_params.capital_base
+        if capital_base <= 0:
+            assert capital_base <= 0, \
```

Comment:
So this is probably unnecessary, since the message inside the error that you're raising is the same as this. Either you probably want that assert, in which case, you're going to get an `AssertionError` with that message, or you want to raise the `ZeroCapitalError`, in which case you could probably pass the text that you wanted displayed to the error class, if you had wanted that to be variable in this context (the thing you're doing with your assert). In this case the assertion is basically serving the same purpose as the if statement, and you don't really need both.

Reply:
I had added the assert because of some logging issues (I wasn't seeing the error) but it resolved. I'll update and remove that